### PR TITLE
Restructure Wallet documentation to promote non-CLI Wallets

### PR DESCRIPTION
**Pull request by danpaul000**
#### Problem
We have no documentation for Trust Wallet or the upcoming Ledger Live support.

#### Summary of Changes
Re-arrange wallet/CLI pages to promote 3rd party wallet solutions.
Add step by step guides for Ledger Live and Trust Wallet.

### CODE REVIEW

**Comment by danpaul000** [comment]
A bit of a big re-arrangement here, please let me know what you think on structure and content.  One thing I did not yet get to in this PR is actually folding in the CLI-based step-by-step google docs I wrote the other day.  More important I think is to make it very clear that TW and Ledger are preferred for most users and give a solid set up for these solutions.  Language currently implies CLI is for more advanced users only, so I didn't think it as urgent to re-factor the CLI-specific content.

File: docs/src/introduction.md [hunk 1]
```diff
@@ -6,11 +6,11 @@ Solana is an open source project implementing a new, high-performance, permissio
 
 ## Why Solana?
```
Comment (danpaul000) [inline docs/src/introduction.md]: I only removed contractions here (can't --> cannot, etc), no content changes.

File: docs/src/wallet/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+The *public key* (commonly shortened to *pubkey*) is known as the wallet's *receiving address* or simply its *address*.  The wallet address can be shared and displayed freely.  When another party is going to send some amount of cryptocurrency to a wallet, they need to know the wallet's receiving address.  Depending on a blockchain's implementation, the address can also be used to view certain information about a wallet, such as viewing the balance, but has no ability to change anything about the wallet or withdraw any tokens.
+
+The *private key* is required to digitally sign any transactions to send cryptocurrencies to another address or to make any changes to the wallet.  The private key must never be shared.  If someone gains access to the private key to a wallet, they can withdraw all the tokens it contains.  If the private key for a wallet is lost, any tokens that have been sent to that wallet's address are permanently lost.
```
Comment (t-nelson) [inline docs/src/wallet/README.md]: I feel like we need some ***emphasis*** on "never be shared" here, but we're already using bold for keywords

File: docs/src/wallet/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+Different wallet solutions are available to manage the compromise between securely storing and backing up the private key and providing a convenient way to interact with the keypair and sign transactions to use/spend the tokens.
+
+Some wallets are easier to use than others.  Some are more secure than others.  Solana [supports multiple types of wallets](supported-wallets.md) from which users can choose.
```
Comment (t-nelson) [inline docs/src/wallet/README.md]: Prehaps we should reorder that last sentence "Multiple types of wallets support Solana..." ?

File: docs/src/SUMMARY.md [hunk 4]
```diff
@@ -1,13 +1,17 @@
...
 * [Introduction](introduction.md)
-* [Install the Solana Tool Suite](install-solana.md)
-* [Command-line Guide](cli/README.md)
-  * [Choose a Wallet](cli/choose-a-wallet.md)
-  * [Hardware Wallets](remote-wallet/README.md)
-    * [Ledger Hardware Wallet](remote-wallet/ledger.md)
-  * [Paper Wallet](paper-wallet/README.md)
-    * [Paper Wallet Usage](paper-wallet/usage.md)
+* [Wallet Guide](wallet/README.md)
+   * [Supported Wallets](wallet/supported-wallets.md)
```
[the remark this answers is not in the document]
Comment (danpaul000) [reply docs/src/SUMMARY.md]: I include a reference in "supported wallets" to our CLI tooling, so it's not entirely Web Wallets

File: docs/src/SUMMARY.md [hunk 5]
```diff
@@ -1,13 +1,17 @@
...
+     * [Trust Wallet](wallet/trust-wallet.md)
+     * [Ledger Nano S](wallet/ledger-live.md)
```
[the remark this answers is not in the document]
Comment (danpaul000) [reply docs/src/SUMMARY.md]: I think we should leave it as Nano S, as that's really the wallet and Live is just a viewer?  Dunno, but I guess I should rename the file to match whichever we decide.

File: docs/src/wallet/README.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+A crypto *wallet* is a scheme that allows users to send, receive, and track ownership of cryptocurrencies.  Wallets can take many forms. A wallet might be a directory or file in your computer's file system, a piece of paper, or a specialized device called a *hardware wallet*.  There are also various smartphone apps and computer programs that can create and manage user-friendly wallets.
+
+A wallet is primarily composed of a *private key* and its cryptographically-derived *public key*.  A private key and its corresponding public key are together known as a *keypair*.  A wallet is simply a way for a user to keep track of and manage one or more keypairs.
+
+The *public key* (commonly shortened to *pubkey*) is known as the wallet's *receiving address* or simply its *address*.  The wallet address can be shared and displayed freely.  When another party is going to send some amount of cryptocurrency to a wallet, they need to know the wallet's receiving address.  Depending on a blockchain's implementation, the address can also be used to view certain information about a wallet, such as viewing the balance, but has no ability to change anything about the wallet or withdraw any tokens.
```
[the remark this answers is not in the document]
Comment (danpaul000) [reply docs/src/wallet/README.md]: Could say the pubkey is "related to" the address?  Or could specify: "On Solana, the pubkey is the address...".  I was trying to keep it pretty generic but it doesn't have to be strict, as long as people get the general idea.

File: docs/src/wallet/README.md (hunk 2 above)
Comment (danpaul000) [reply docs/src/wallet/README.md]: Single asterisk is italics for keywords, double asterisks for important stuff, like this.  I'll add it.

File: docs/src/SUMMARY.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (t-nelson) [reply docs/src/SUMMARY.md]: Sorry I meant to come back and leave a comment for rational on this comment and the next

Idea behind this organizational change is that TW and LL will be similar user experiences.  Grouping them together seems logical.

File: docs/src/SUMMARY.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (t-nelson) [reply docs/src/SUMMARY.md]: The Nano S device can be supported by many SW solutions.  So having a high-level entry point named after will likely create confusion. (eg. I'd used my Nano S with the CLI in the past, comeback looking for instructions and the most obvious title points me to a very different guide)

File: docs/src/wallet/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (t-nelson) [reply docs/src/wallet/README.md]: The pubkey has another, arguably more important, role in that it is what the network uses to confirm that you have authority to move coins.  Perhaps we could split out the first part introducing the pubkey concept from that angle.  Then segway into the address concept

**Comment by garious** [comment]
Because CI is super fast when making markdown changes and because Gitbook is awful, you should make docs PRs small and easily revertable.  In this PR particularly, you move the install-solana page, which will probably break Gitbook in the same way that I broke it last time I made that change.

As for the rest, removing contractions and line formatting can be merged independently and without review.  Adding *more* docs within the existing structure is easy and Trent's feedback there is plenty sufficient.  Restructuring changes, however, need to be carefully reviewed and should be done in a standalone review.

At a high-level, I like the idea of restructuring to be inclusive of app-based wallets. Just need to get there in smaller steps.

**Comment by danpaul000** [comment]
Closing this PR to follow up with smaller PRs, one on structure, one on content.

**Pull request closed without merging**